### ISSUE TITLE
Segfault when a PyCharm debug breakpoint is set within pygame event handling

### ISSUE THREAD

**Issue by BelTol3011** [pygame/pygame#3456]
**Environment:**
- **Operating system**: Linux(Manjaro)
- **Python version**: 3.10.6
- **SDL version**: 2.0.16
- **PyGame version**: 2.1.0

**Current behavior:**
When I set a breakpoint in PyCharm like shown below (see the red dot), the program terminates in a segfault after running it with the debugger in PyCharm. I wasn't sure wether to post this on the PyCharm bug tracker or the pygame one. I chose pygame sice the segfault happens inside the pygame library (supposedly).

**Expected behavior:**
I expect no segfault and the python debugger to halt when the line with the breakpoint is reached. This correctly happens if instead the breakpoint is invoked by `breakpoint()`.

**Screenshots**
![image](https://user-images.githubusercontent.com/37810842/190869016-51571a30-f567-4dc7-88d1-751ac1607f40.png)

**Steps to reproduce:**
1. Create a new file in PyCharm and paste the test code
2. Set a breakpoint on line 7 by clicking to the left of the line
3. Debug the program by clicking on the green bug / pressing SHIFT-F9

**Test code**
```py
import pygame

pygame.init()

for event in pygame.event.get():
    if event.type == pygame.QUIT:
        assert True  # a random statement to be able to set a breakpoint

```

**Stack trace/error output/other error logs**
```
pygame 2.1.0 (SDL 2.0.16, Python 3.10.6)
Hello from the pygame community. https://www.pygame.org/contribute.html
Fatal Python error: pygame_parachute: (pygame parachute) Segmentation Fault
Python runtime state: initialized

Thread 0x00007f05e71f96c0 (most recent call first):
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/pydevd.py", line 138 in _on_run
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 219 in run
  File "/usr/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/usr/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x00007f05e79fa6c0 (most recent call first):
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 293 in _on_run
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 219 in run
  File "/usr/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/usr/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x00007f05e81fb6c0 (most recent call first):
  File "/usr/lib/python3.10/threading.py", line 324 in wait
  File "/usr/lib/python3.10/queue.py", line 180 in get
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 368 in _on_run
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 219 in run
  File "/usr/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/usr/lib/python3.10/threading.py", line 973 in _bootstrap

Current thread 0x00007f0609f5e740 (most recent call first):
  File "/home/bela/.config/JetBrains/PyCharm2022.2/scratches/scratch_5.py", line 6 in <module>
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18 in execfile
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/pydevd.py", line 1496 in _exec
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/pydevd.py", line 1489 in run
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/pydevd.py", line 2177 in main
  File "/home/bela/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/222.4167.33/plugins/python/helpers/pydev/pydevd.py", line 2195 in <module>

Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg.lapack_lite, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, matplotlib._c_internal_utils, PIL._imaging, matplotlib._path, kiwisolver, _pydevd_bundle.pydevd_cython, _pydevd_bundle_ext.pydevd_cython, _pydevd_frame_eval.pydevd_frame_evaluator_common, _pydevd_frame_eval_ext.pydevd_frame_evaluator_common, _pydevd_frame_eval.pydevd_frame_evaluator_py39_and_above, _pydevd_frame_eval_ext.pydevd_frame_evaluator_py39_and_above, pygame.base, pygame.constants, pygame.rect, pygame.rwobject, pygame.surflock, pygame.color, pygame.bufferproxy, pygame.math, pygame.surface, pygame.display, pygame.draw, pygame.event, pygame.imageext, pygame.image, pygame.joystick, pygame.key, pygame.mouse, pygame.time, pygame.mask, pygame.pixelcopy, pygame.transform, pygame.font, pygame.mixer_music, pygame.mixer, pygame.scrap, pygame.fastevent, pygame._freetype (total: 51)

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

**Comment by novialriptide** [comment]
I tested this on my machine and it works completely fine. Try upgraading to pygame 2.1.2

**Comment by BelTol3011** [comment]
I have tried updating the pygame version to 2.1.2 and creating a fresh venv but the issue persisted. Then I tried installing a new version of PyCharm and it worked! After installing the Cython debug extension that speeds up PyCharm debugging it stopped working. I read a bit and it seems that segfault issues associated with this extension (and breakpoints) are not rare. I think it's safe to assume that this is not a problem with pygame but rather the Cython PyCharm extension.
